File: python/euler/bin/p0006.py
# project euler: problem 6


def compute(num: int) -> str:
    sum_of_squares = sum(i * i for i in range(1, num + 1))
    square_of_sum = pow((1 + num) * num // 2, 2)

    # The square of sum is equal or larger than the sum of squares.
    return str(square_of_sum - sum_of_squares)


def solve() -> str:
    return compute(100)

File: typescript/euler/bin/p0006.ts
// project euler: problem 6

import { sum } from "../lib/math.ts";
import { range } from "../lib/util.ts";

export const compute = (upper: number): string => {
  const sequence = range(1, upper + 1);

  // The square of sum is equal or larger than the sum of squares.
  return String(sum(sequence) ** 2 - sum(sequence.map((n) => n * n)));
};

export const solve = (): string => compute(100);
